wallets/react/src/hub/utils.ts

```diff
@@ -0,0 +1,388 @@
+        maybeNetwork,
+        coreState,
+        eventInfo
+      );
+    }
+  });
+}
+
+/**
+ * For backward compatibility, there is an special namespace called DISCOVER_MODE.
+ * Alongside `DISCOVER_MODE`, `network` will be set as well. here we are manually matching networks to namespaces.
+ * This will help us keep the legacy interface and have what hub needs as well.
+ */
+export function discoverNamespace(network: string): Namespace {
+  // This trick is using for enforcing exhaustiveness check.
+  network = network as unknown as Networks;
+  switch (network) {
+    case Networks.AKASH:
+    case Networks.BANDCHAIN:
+    case Networks.BITCANNA:
+    case Networks.BITSONG:
+    case Networks.BINANCE:
+    case Networks.CRYPTO_ORG:
+    case Networks.CHIHUAHUA:
+    case Networks.COMDEX:
+    case Networks.COSMOS:
+    case Networks.CRONOS:
+    case Networks.DESMOS:
+    case Networks.EMONEY:
+    case Networks.INJECTIVE:
+    case Networks.IRIS:
+    case Networks.JUNO:
+    case Networks.KI:
+    case Networks.KONSTELLATION:
+    case Networks.KUJIRA:
+    case Networks.LUMNETWORK:
+    case Networks.MEDIBLOC:
+    case Networks.OSMOSIS:
+    case Networks.PERSISTENCE:
+    case Networks.REGEN:
+    case Networks.SECRET:
+    case Networks.SENTINEL:
+    case Networks.SIF:
+    case Networks.STARGAZE:
+    case Networks.STARNAME:
+    case Networks.TERRA:
+    case Networks.THORCHAIN:
+    case Networks.UMEE:
+      return Namespace.Cosmos;
+    case Networks.AVAX_CCHAIN:
+    case Networks.ARBITRUM:
+    case Networks.BOBA:
+    case Networks.BSC:
+    case Networks.FANTOM:
+    case Networks.ETHEREUM:
+    case Networks.FUSE:
+    case Networks.GNOSIS:
+    case Networks.HARMONY:
+    case Networks.MOONBEAM:
+    case Networks.MOONRIVER:
+    case Networks.OPTIMISM:
+    case Networks.POLYGON:
+    case Networks.STARKNET:
+      return Namespace.Evm;
+    case Networks.SOLANA:
+      return Namespace.Solana;
+    case Networks.BTC:
+    case Networks.BCH:
+    case Networks.DOGE:
+    case Networks.LTC:
+    case Networks.TRON:
+      return Namespace.Utxo;
+    case Networks.POLKADOT:
+    case Networks.TON:
+    case Networks.AXELAR:
+    case Networks.MARS:
+    case Networks.MAYA:
+    case Networks.STRIDE:
+    case Networks.Unknown:
+      throw new Error("Namespace isn't supported. network: " + network);
+  }
+
+  throw new Error(
+    "Couldn't matched network with any namespace. it's not discoverable. network: " +
+      network
+  );
+}
+
+export function getLegacyProvider(
+  allProviders: VersionedProviders[],
+  type: string
+): LegacyProviderInterface {
+  const [legacy] = separateLegacyAndHubProviders(allProviders);
+  const provider = legacy.find((legacyProvider) => {
+    return legacyProvider.config.type === type;
+  });
+
+  if (!provider) {
+    console.warn(
+      `You have a provider that hasn't legacy provider. it causes some problems since we need some legacy functionality. Provider Id: ${type}`
```

Comment:
```suggestion
      `You have a provider that doesn't have legacy provider. It causes some problems since we need some legacy functionality. Provider Id: ${type}`
```

Reply:
updated.